workers/unity/Assets/Gdk/Core/GameObjectRepresentation/GameObjectDispatcherCallbacksSystem.cs

```diff
@@ -0,0 +1,70 @@
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Gathers incoming dispatcher ops and invokes callbacks on relevant GameObjects.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    public class GameObjectDispatcherCallbacksSystem : ComponentSystem
+    {
+        private MutableView view;
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            var worker = WorkerRegistry.GetWorkerForWorld(World);
+            view = worker.View;
+
+            GenerateComponentGroups();
+        }
+
+        private void GenerateComponentGroups()
+        {
+            foreach (var gameObjectTranslation in view.GameObjectTranslations)
+            {
+                gameObjectTranslation.ComponentAddedComponentGroup =
+                    GetComponentGroup(gameObjectTranslation.ComponentAddedComponentTypes);
+                gameObjectTranslation.ComponentRemovedComponentGroup =
+                    GetComponentGroup(gameObjectTranslation.ComponentRemovedComponentTypes);
+                gameObjectTranslation.AuthoritiesChangedComponentGroup =
+                    GetComponentGroup(gameObjectTranslation.AuthoritiesChangedComponentTypes);
+                if (gameObjectTranslation.ComponentsUpdatedComponentTypes.Length > 0)
+                {
+                    gameObjectTranslation.ComponentsUpdatedComponentGroup =
+                        GetComponentGroup(gameObjectTranslation.ComponentsUpdatedComponentTypes);
+                }
+
+                gameObjectTranslation.EventsReceivedComponentGroups =
+                    new ComponentGroup[gameObjectTranslation.EventsReceivedComponentTypeArrays.Length];
+                for (var i = 0; i < gameObjectTranslation.EventsReceivedComponentTypeArrays.Length; i++)
+                {
+                    gameObjectTranslation.EventsReceivedComponentGroups[i] =
+                        GetComponentGroup(gameObjectTranslation.EventsReceivedComponentTypeArrays[i]);
+                }
+
+                gameObjectTranslation.CommandRequestsComponentGroups =
+                    new ComponentGroup[gameObjectTranslation.CommandRequestsComponentTypeArrays.Length];
+                for (var i = 0; i < gameObjectTranslation.CommandRequestsComponentTypeArrays.Length; i++)
+                {
+                    gameObjectTranslation.CommandRequestsComponentGroups[i] =
+                        GetComponentGroup(gameObjectTranslation.CommandRequestsComponentTypeArrays[i]);
+                }
+            }
+        }
+
+        protected override void OnUpdate()
+        {
+            foreach (var gameObjectTranslation in view.GameObjectTranslations)
+            {
+                gameObjectTranslation.InvokeOnAddComponentCallbacks();
```

Comment:
Are we concerned about the fact that callbacks are technically called out of order here? (Compared to how the ops are received)

Reply:
Yeah, this is a limitation of the GDK Core since by the time the reactive components are added to the ecs entities, you have no information about the order at which the ops came in anymore. This might lead to breaks in legacy projects.

Reply:
Gotcha, we will need to make sure this is communicated properly :)

Reply:
No, all callbacks are invoked after the data model was updated. By the time, the callbacks for an update for A is fired all component values are up-to-date.